pkg/controller/directvolumemigrationprogress/directvolumemigrationprogress_controller.go

```diff
@@ -276,6 +281,12 @@ func (r *ReconcileDirectVolumeMigrationProgress) reportContainerStatus(pvProgres
 		exitCode := containerStatus.LastTerminationState.Terminated.ExitCode
 		pvProgress.Status.ExitCode = &exitCode
 		pvProgress.Status.ContainerElapsedTime = &metav1.Duration{Duration: containerStatus.LastTerminationState.Terminated.FinishedAt.Sub(containerStatus.LastTerminationState.Terminated.StartedAt.Time).Round(time.Second)}
+	case !containerStatus.Ready && containerStatus.State.Waiting != nil && containerStatus.State.Waiting.Reason == ContainerCreating:
+		// if pod is not in running state after 10 mins of its creation, assume the pod is in failed state
+		if time.Now().Sub(pod.CreationTimestamp.Time) > 10*time.Minute {
+			pvProgress.Status.PodPhase = kapi.PodFailed
```

Comment:
do we really want this? 

Ideally, we should raise a warning right? that says that a pod is in creating the state since 10 mins?

Reply:
+1 to rasing warning